### ISSUE TITLE
Raw strings don't close properly

### ISSUE THREAD

**Issue by KAAtheWiseGit** [nushell/nushell#13758]
### Describe the bug

I was doing a pattern search and encountered a raw string, which didn't work:

```Nushell
~> rg --multiline-dotall `<span class="math inline">\(`
```

Nushell treats it as if it wasn't closed.

### How to reproduce

-

### Expected behavior

-

### Configuration

| key                | value                                                        |
| ------------------ | ------------------------------------------------------------ |
| version            | 0.97.1                                                       |
| major              | 0                                                            |
| minor              | 97                                                           |
| patch              | 1                                                            |
| branch             |                                                              |
| commit_hash        |                                                              |
| build_os           | linux-x86_64                                                 |
| build_target       | x86_64-alpine-linux-musl                                     |
| rust_version       | rustc 1.80.1 (3f5fd8dd4 2024-08-06) (Alpine Linux 1.80.1-r0) |
| cargo_version      | cargo 1.80.1 (376290515 2024-07-16)                          |
| build_time         | 2024-08-29 21:40:07 +00:00                                   |
| build_rust_channel | release                                                      |
| allocator          | mimalloc                                                     |
| features           | default, sqlite, system-clipboard, trash                     |
| installed_plugins  | endecode 0.2.0, template 0.1.0                               |

**Comment by 1Kinoti** [comment]
I think this is probably because of the `(` at the end, which is an interpolation.

I wonder why interpolations are accepted in these kinds of strings

**Comment by fdncred** [comment]
That's also not a nushell raw string.

**Comment by KAAtheWiseGit** [comment]
Right, backtick quotes.  I'm starting to think there might be too many quoting methods in Nushell

**Comment by KAAtheWiseGit** [comment]
Another example I stumbled into today:

```Nushell
~/fork/beast-dist/examples/nexus> search --no-ignore-vcs `\(\(`
	# cursor here, as if we're still inside a string
```

**Comment by fdncred** [comment]
there's also some weirdness that i've seen in the past with escaping not quite working right
